sql/src/test/java/io/crate/integrationtests/PartitionedTableIntegrationTest.java

```diff
@@ -175,16 +173,18 @@ public void testCopyFromIntoPartitionedTableWithGeneratedColumnPK() throws Excep
     @Test
     public void testInsertIntoClosedPartition() throws Exception {
         execute("create table t (n integer) partitioned by (n)");
-        ensureYellow();
         execute("insert into t (n) values (1)");
+        refresh();
+        ensureYellow();
 
-        String[] indices = client().admin().indices().getIndex(new GetIndexRequest()).actionGet().getIndices();
-        client().admin().indices().close(new CloseIndexRequest(indices[0])).actionGet();
+        execute("alter table t partition (n = 1) close");
         execute("insert into t (n) values (100)");
-        ensureYellow();
+        execute("insert into t (n) values (1)");
+        refresh();
```

Comment:
I'd move that down after the assertion as it is related to the next select query only